src/inspector_agent.cc

```diff
@@ -706,6 +751,56 @@ void Url(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(OneByteString(env->isolate(), url.c_str()));
 }
 
+static void AsyncTaskScheduledWrapper(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsString())
+    return env->ThrowTypeError("first argument must be a string");
+  if (!args[1]->IsNumber())
+    return env->ThrowTypeError("second argument must be a number");
+  if (!args[2]->IsBoolean())
+    return env->ThrowTypeError("third argument must be a boolean");
```

Comment:
These errors should be never seen by the user as long as async hooks work correctly. These methods are invoked by our own async hook listener from the methods invoked by async hooks (`init`, `before`, etc.). 

> (a) it would be better if the type checking were done in js land if possible

The values are coming from the async_hook module. I could add a type check to my async hook callbacks (`init`, `before`, etc), but that seems too defensive to me - do we really need to worry about async_hooks not honoring their API contract?

Even if we check the types in js land, I think it's still a good idea to check them in C++ land too, so that we don't crash the process (I assume that V8 will crash the process when we try to cast a JS value to an incompatible C++ type)..

> (b) it would be better if they used named arguments rather than "first argument", "second argument", etc

Good point, I'll reword the messages 👍